legion_core/src/world.rs

```diff
@@ -731,6 +731,33 @@ impl World {
     /// Determines if the given `Entity` is alive within this `World`.
     pub fn is_alive(&self, entity: Entity) -> bool { self.entity_allocator.is_alive(entity) }
 
+    /// Returns the entity's component types, if the entity exists.
+    pub fn entity_component_types(
+        &self,
+        entity: Entity,
+    ) -> Option<&[(ComponentTypeId, ComponentMeta)]> {
```

Comment:
If this tuple is used in more places we could also create a `type ComponentType = (ComponentTypeId, ComponentMeta);`